numba_dpex/core/kernel_interface/func.py

```diff
@@ -6,100 +6,217 @@
 """
 
 
+from numba.core import sigutils, types
 from numba.core.typing.templates import AbstractTemplate, ConcreteTemplate
 
+from numba_dpex import config
+from numba_dpex.core.caching import LRUCache, NullCache, build_key
 from numba_dpex.core.compiler import compile_with_dpex
 from numba_dpex.core.descriptor import dpex_target
+from numba_dpex.utils import npytypes_array_to_dpex_array
 
 
-def compile_func(pyfunc, return_type, args, debug=None):
-    cres = compile_with_dpex(
-        pyfunc=pyfunc,
-        pyfunc_name=pyfunc.__name__,
-        return_type=return_type,
-        target_context=dpex_target.target_context,
-        typing_context=dpex_target.typing_context,
-        args=args,
-        is_kernel=False,
-        debug=debug,
-    )
-    func = cres.library.get_function(cres.fndesc.llvm_func_name)
-    cres.target_context.mark_ocl_device(func)
-    devfn = DpexFunction(cres)
+class DpexFunction(object):
+    """Class to materialize dpex function"""
```

Comment:
The description can be improved. How about:

"Helper class to eager compile a specialized `numba_dpex.func` decorated Python function into a LLVM function with `spir_func` calling convention."